sklearn/model_selection/_split.py

```diff
@@ -284,7 +284,16 @@ def __init__(self, n_splits, shuffle, random_state):
 
         self.n_splits = n_splits
         self.shuffle = shuffle
+        # For repr
         self.random_state = random_state
+        if random_state is None:
+            # This is done to ensure that the multiple calls to split
+            # are random for each initialization of splitter but consistent
+            # across multiple calls for the same initialization.
+            self._random_state = check_random_state(
+                random_state).randint(np.iinfo(np.int32).max)
```

Comment:
Doing `check_random_state(self.random_state).get_state()` in all cases will ensure similar behaviour regardless of the form of the parameter.